lib/KeyboardAwareHOC.js

```diff
@@ -430,6 +441,141 @@ function KeyboardAwareHOC(
+      const viewableAreaStartY = this.topDistanceToWindow;
+      const viewableAreaEndY = keyboardEvent.endCoordinates.screenY - this.props.inputAccessoryViewHeight;
+      const viewableAreaHeight = viewableAreaEndY - viewableAreaStartY;
+      return viewableAreaHeight > distanceBetweenCaretAndBottomOfField;
+    }
+
+    _isCaretUnderKeyboard(caretY: number, caretHeight: number, keyboardEvent: Object) {
+      const caretBottomPosition = caretY + caretHeight;
+      const keyboardPosition = keyboardEvent.endCoordinates.screenY
+      return caretBottomPosition > (keyboardPosition - this.props.inputAccessoryViewHeight);
+    }
+
+    _isCaretAboveViewableArea(caretY: number) { //caret may remain above the viewable area when orientation changes
+      return this.topDistanceToWindow > caretY;
+    }
+
+    _isCaretOutOfViewableArea(caretY: number, caretHeight: number, keyboardEvent: Object) {
+      return this._isCaretUnderKeyboard(caretY, caretHeight, keyboardEvent) || 
+            this._isCaretAboveViewableArea(caretY);
+    }
+    
+    _shouldScrollToBottomOfComponent(fieldY: number, fieldHeight: number, caretY: number, caretHeight: number, textInputBottomTextInset: number, keyboardEvent: Object) 
+    { //is there enough place in the viewable area when keyboard is open?
+      return this._isFocusedAreaFitsInViewableArea( caretY, fieldY, fieldHeight, keyboardEvent) && 
+            //is the caret at the last line of the component?
+            this._isCaretAtLastLine(caretY, caretHeight, fieldY, fieldHeight, textInputBottomTextInset); 
+    }
+
+    _scrollToBottomOfComponent(currentlyFocusedField: number, keyboardEvent: Object, keyboardSpace: number) {
+      UIManager.measureInWindow(
+        currentlyFocusedField,
+        (x: number, y: number, width: number, height: number) => {
+          this._doScrollToBottomOfComponent(currentlyFocusedField, y, height, keyboardEvent, keyboardSpace);
+        }
+      );
+    }
+
+    _extraScrollHeight() {
+      return this.props.extraScrollHeight + this._totalVerticalDistanceToWindow();
+    }
+
+    _doScrollToBottomOfComponent(currentlyFocusedField: number, y: number, height: number, keyboardEvent: Object, keyboardSpace: number) {
+      // Check if the TextInput will be hidden by the keyboard
+      const textInputBottomPosition = y + height
+      const keyboardPosition = keyboardEvent.endCoordinates.screenY
+      const totalExtraHeight =
+        this._extraScrollHeight() + this.props.extraHeight
+      if (Platform.OS === 'ios') {
+        if (
+          textInputBottomPosition >
+          keyboardPosition - totalExtraHeight
+        ) {
+          this._scrollToFocusedInputWithNodeHandle(
+            currentlyFocusedField
+          )
+        }
+      } else {
+        // On android, the system would scroll the text input just
+        // above the keyboard so we just neet to scroll the extra
```

Comment:
Small typo 
```suggestion
        // above the keyboard so we just need to scroll the extra
```

Reply:
Fixed https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/2/commits/6ba99f4f419b306d2bfaac7f6d6ea22abfa40169

Reply:
Cool! 
Just being curious, does the GitHub `Suggested change` thing works?

Reply:
`Suggested change` works actually, it just sometimes doesn't catch my attention :)

Reply:
I fixed that manually ;)